addons/pandora/ui/components/properties/reference/reference_property.gd

```diff
@@ -27,40 +22,18 @@ func _ready() -> void:
 
 func refresh() -> void:
 	if _property != null:
-		entity_picker.set_filter(_get_setting(CATEGORY_FILTER) as String)
-		entity_picker.categories_only = _get_setting(CATEGORIES_ONLY) as bool
-		match _get_setting(SORT_LIST) as String:
-			SORT_ALPHABETICALLY:
+		entity_picker.set_filter(_property.get_setting(ReferenceType.SETTING_CATEGORY_FILTER) as String)
+		entity_picker.categories_only = _property.get_setting(ReferenceType.SETTING_CATEGORIES_ONLY) as bool
+		match _property.get_setting(ReferenceType.SETTING_SORT_LIST) as String:
+			ReferenceType.SORT_ALPHABETICALLY:
 				entity_picker.set_sort(func(a,b): return a.get_entity_name() < b.get_entity_name())
-			SORT_AS_IS:
+			ReferenceType.SORT_AS_IS:
 				entity_picker.set_sort(func(a,b): return false)
 		var entity = _property.get_default_value() as PandoraEntity
 		if entity != null:
 			entity_picker.select.call_deferred(entity)
-
-
-
-func get_default_settings() -> Dictionary:
```

Comment:
Settings are now comping from the type directly, rather than from some random UI component.